pkg/bcr/registry.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2023 OnMetal authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bcr
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"k8s.io/apimachinery/pkg/util/yaml"
+
+	ori "github.com/onmetal/onmetal-api/ori/apis/bucket/v1alpha1"
+)
+
+func LoadBucketClasses(reader io.Reader) ([]ori.BucketClass, error) {
+	var classList []ori.BucketClass
+	if err := yaml.NewYAMLOrJSONDecoder(reader, 4096).Decode(&classList); err != nil {
+		return nil, fmt.Errorf("unable to unmarshal bucket classes: %w", err)
+	}
+
+	return classList, nil
+}
+
+func LoadBucketClassesFile(filename string) ([]ori.BucketClass, error) {
+	file, err := os.Open(filename)
```

Comment:
You need to close the file handle at some point by calling `file.Close()`. Ideally this has to be done via a `defer` call.

Reply:
Done